### PR TITLE
add Auth function(middleware) and modify the path of import like "bookcommunity/modle"

### PR DESCRIPTION
**Pull request by shiminjia**

### CODE REVIEW

File: controller/user.go [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
 import (
-	"net/http"
-
+	"bookcommunity/config"
+	"bookcommunity/middleware"
+	"bookcommunity/model"
 	"github.com/gin-gonic/gin"
-	"github.com/shiminjia/bookcommunity/config"
-	"github.com/shiminjia/bookcommunity/middleware"
-	"github.com/shiminjia/bookcommunity/model"
```
Comment (ulip128) [inline controller/user.go]: Dont't remove ulip128/bookcommunity.

File: middleware/auth.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+func AuthRequired() gin.HandlerFunc {
+	return func(c *gin.Context){
+		token := c.GetHeader("Authorization")
+		if token == "" {
+			ErrorResponse(c, http.StatusUnauthorized, config.UnverifiedError_1)
+			c.Abort()
+			return
+		}
+		if token != "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1MjgwMTY5MjIsImlkIjowLCJuYmYiOjE1MjgwMTY5MjIsInVzZX"+
```
Comment (ulip128) [inline middleware/auth.go]: Don't write an authentication token directly.

**Comment by ulip128** [comment]
Use go fmt command.

File: middleware/auth.go (hunk 2 above)
Comment (shiminjia) [reply middleware/auth.go]: it is just a simple

Comment (shiminjia) [reply middleware/auth.go]: l have not written JTW function now. So l wrote authentication directly for api test.

File: controller/user.go (hunk 1 above)
Comment (shiminjia) [reply controller/user.go]: l am using the relative path to import module which l wrote.
No need to write path like "github.com/shiminjia/bookcommunity/middleware" any more.

**Pull request merged**